CriteoPublisherSdk/Tests/UnitTests/CR_BidManagerTests.m

```diff
@@ -350,65 +667,177 @@ - (void)testLiveBid_GivenNoBid_ThenNoResponseGiven {
   CR_CdbBid *noBid = CR_CdbBidBuilder.new.adUnit(self.adUnit1).noBid().build;
   [self givenApiHandlerRespondBid:noBid];
 
+  [self fetchLiveBidAndExpectBidCached:nil bidConsumedAndResponded:self.bid1];
+
   // No bid from cdb call has not been cached
-  OCMReject([self.cacheManager setBid:[OCMArg any]]);
+  XCTAssertNotEqual(self.cacheManager.bidCache[self.adUnit1], noBid);
+}
 
-  [self.bidManager fetchLiveBidForAdUnit:self.adUnit1
-                         responseHandler:^(CR_CdbBid *bid) {
-                           XCTAssertNil(bid);
-                         }];
-  CR_OCMockVerifyCallCdb(self.apiHandlerMock, @[ self.adUnit1 ]);
+- (void)testLiveBid_GivenNoBidAfterTimeBudget_ThenNoResponseGiven {
+  CR_CdbBid *noBid = CR_CdbBidBuilder.new.adUnit(self.adUnit1).noBid().build;
+  [self givenApiHandlerRespondBid:noBid];
+  [self givenTimeBudgetExceeded];
+
+  [self fetchLiveBidAndExpectBidCached:noBid bidConsumedAndResponded:self.bid1];
 
   // No bid from cdb call has not been cached
   XCTAssertNotEqual(self.cacheManager.bidCache[self.adUnit1], noBid);
 }
 
+- (void)testLiveBid_GivenConcurrentCalls_ThenBidsFromResponsesGivenWithOrder {
```

Comment:
Nit: the order is not important. The important is that concurrency exists.
With cache bidding, an ad unit can only be fetched once at the same time. With live bidding, there is no limitation.

Reply:
Ok, the order there is used to check that concurrency is properly handled and not FIFO, an interpretation of "at the same time" and to prevent queueing issues.